conan/tools/scm/git.py

```diff
@@ -22,7 +22,7 @@ def get_commit(self):
             # --full-history is needed to not avoid wrong commits:
             # https://github.com/conan-io/conan/issues/10971
             # https://git-scm.com/docs/git-rev-list#Documentation/git-rev-list.txt-Defaultmode
-            commit = self._run('rev-list HEAD -n 1 --full-history -- "{}"'.format(self.folder))
+            commit = self._run('rev-list HEAD -n 1 --full-history -- "."')
```

Comment:
But this change will not guarantee that this is run in the specified ``self.folder``, but in the ``cwd``, which is not guarantee that it will be ``self.folder``, if users specify a different value in the constructor, which they can.

Reply:
Oh, yes, you are right, I missed that part. Looks good to me then.